src/smt/theory_str_noodler/length_decision_procedure.h

```diff
@@ -0,0 +1,258 @@
+#ifndef _NOODLER_LENGTH_DECISION_PROCEDURE_H_
+#define _NOODLER_LENGTH_DECISION_PROCEDURE_H_
+
+#include <memory>
+#include <deque>
+#include <algorithm>
+
+#include "smt/params/theory_str_noodler_params.h"
+
+#include "formula.h"
+#include "inclusion_graph.h"
+#include "aut_assignment.h"
+#include "formula_preprocess.h"
+#include "decision_procedure.h"
+
+namespace smt::noodler {
+
+    /**
+     * @brief Pool of variable constraints
+     * 
+     */
+    class ConstraintPool : public std::map<BasicTerm, class VarConstraint> {
```

Comment:
Does this compile? Because `VarConstraint` is defined after this.
Maybe I would move this after the `VarConstraint` definition anyway.